### PR TITLE
Rotate haskell-pushbot keys

### PR DESCRIPTION
**Pull request by ezyang**
Signed-off-by: Edward Z. Yang <ezyang@cs.stanford.edu>

Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/#conventions).
* [x] Any changes that could be relevant to users have been recorded in the changelog.
* [x] The documentation has been updated, if necessary.
* [x] If the change is docs-only, `[ci skip]` is used to avoid triggering the build bots.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

**Comment by phadej** [comment]
Why can't we use https://docs.travis-ci.com/user/encryption-keys/ and not play games?

**Comment by phadej** [comment]
FWIW, my `id_ed25519` is 444 bytes long. That one can live in env var directly.

**Comment by phadej** [comment]
Ah I see why we cannot use secrets. Is there a comment about that somewhere in the code?

**Comment by 23Skidoo** [comment]
Looks like there's a bug in the bit where we inoke `rot13`:

```
+rot13
./travis/upload.sh: line 42: id_rsa: No such file or directory
```

File: travis/upload.sh [hunk 1]
```diff
@@ -39,7 +39,7 @@ cd travis
 # Setup SSH key we will use to push to binaries repository.
 # umask to get the permissions to be 600 (not 400, because the deploy
 # script in .travis.yml is going to clobber this private key)
-(umask 177 && cp id_rsa $HOME/.ssh/id_rsa)
+(umask 177 && rot13 < id_rsa > $HOME/.ssh/id_rsa)
```
Comment (23Skidoo) [inline travis/upload.sh]: Shouldn't the input file be `id_rsa.rot13`?

File: travis/upload.sh [hunk 2]
```diff
@@ -65,7 +65,7 @@ mkdir cabal-install
 cp -R $TRAVIS_BUILD_DIR/Cabal/tests                                  Cabal
 cp -R $TRAVIS_BUILD_DIR/cabal-install/tests                          cabal-install
 # Copy in credentials so we can delete branch when done
-cp $TRAVIS_BUILD_DIR/travis/id_rsa .
+rot13 < $TRAVIS_BUILD_DIR/travis/id_rsa > id_rsa.rot13
```
Comment (23Skidoo) [inline travis/upload.sh]: Here as well.

**Comment by 23Skidoo** [comment]
Pushed a fix, let's see if it unbreaks things.

**Comment by ezyang** [comment]
I am a bit perplexed by the JMU errors. What's going on here?

**Comment by 23Skidoo** [comment]
Looks like the usual intermittent Travis flakiness, I've restarted the build.

**Comment by quasicomputational** [comment]
I've tried to fix it. Let's see what Travis thinks...

**Comment by 23Skidoo** [comment]
Uhh, maybe that's because we're pushing the un-ROT13ed key to a branch in the `cabal-binaries` repo, and GitHub is smart enough to notice that.

**Comment by quasicomputational** [comment]
That sounds very plausible. Does the cabal-binaries repo actually need the key? If it does, we can just avoid putting it in the clear (since we deliberately do that now), and then whatever it is that needs the key will need to learn to 'decrypt' it. Otherwise, we can just delete that line and go on happily.

Either way I think @ezyang needs to generate a fresh key and install it on the haskell-pushbot account, right?

**Comment by ezyang** [comment]
LOL. I thought I was pushing the rot13'ed key; apparently not. We do need to push the key with the commit because the downstream jobs are responsible for deleting their binaries when they fail. Issuing new key...

**Comment by 23Skidoo** [comment]
Yes, I think we can push a ROT13-ed key to cabal-binaries and then decrypt it during the build step.

**Comment by ezyang** [comment]
Looking good.

**Comment by 23Skidoo** [comment]
The 7.10.3 failure is an unrelated issue, otherwise everything is green.

**Comment by 23Skidoo** [comment]
Merged, thanks!

**Pull request merged**